Switch to ansi_up for ansi rendering in actions

silverwind opened the pull request:
Fixes: https://github.com/go-gitea/gitea/issues/24777

<img width="616" alt="Screenshot 2023-06-20 at 23 08 33" src="https://github.com/go-gitea/gitea/assets/115237/5a7e34fc-c071-40de-90d9-3ae417ebac75">
<img width="559" alt="Screenshot 2023-06-20 at 23 08 22" src="https://github.com/go-gitea/gitea/assets/115237/d8fe682b-7853-4e8b-b8ed-f98f3d33d673">
<img width="509" alt="Screenshot 2023-06-20 at 23 08 45" src="https://github.com/go-gitea/gitea/assets/115237/db47bf47-241c-4f78-bb95-ea69dcaacdf5">

delvh commented on web_src/js/components/RepoActionView.vue:
```diff
@@ -119,13 +119,13 @@
 import {SvgIcon} from '../svg.js';
 import ActionRunStatus from './ActionRunStatus.vue';
 import {createApp} from 'vue';
-import AnsiToHTML from 'ansi-to-html';
```
I think you are currently breaking the test for this file.

silverwind replied:
Yes, the tests need some updates, I'm a bit lost in them currently.

silverwind replied:
Tests are passing, still need some verification if results are ok.

delvh reviewed (approved):
But apart from that LGTM

silverwind commented:
@wxiaoguang can you help me update these tests? They talk about some bugs in AnsiToHTML but those seem not present in `ansi_up` from cursory testing.

silverwind commented:
Tests are now passing, but I'm not 100% certain the results are as expected. There are a number of `style="color:rgb(255,255,255)"` outputs, but actions output itself does not have this white color for uncolored text which seems odd.

silverwind commented on web_src/js/components/RepoActionView.test.js:
```diff
@@ -11,20 +12,29 @@ test('processConsoleLine', () => {
   // treat "\033[0K" and "\033[0J" (Erase display/line) as "\r", then it will be covered to "\n" finally.
-  expect(ansiLogToHTML('a\x1b[Kb\x1b[2Jc')).toEqual('a\nb\nc');
+  expect(ansiLogToHTML('a\x1b[Kb\x1b[2Jc')).toEqual(
+    '<span style="color:rgb(255,255,255)">a</span><span style="color:rgb(255,255,255)">b</span><span style="color:rgb(255,255,255)">c</span>'
```
Newlines gone in this test case because I removed `ansiRegexpNewLine`. Wonder whether to restore.

wxiaoguang replied:
It needs to keep the newlines.

Otherwise some progress output are rendered in one line which makes a pretty long line like:

```
Download .... 1% Download .... 10% Download .... 50% Download .... 99% Download .... 100%
```


With the new line, the expected output is:

```
Download .... 1%
Download .... 10%
Download .... 50%
Download .... 99%
Download .... 100%`
```


Context: https://github.com/go-gitea/gitea/pull/23820

silverwind replied:
I will restore it, thanks

silverwind replied:
Restored in https://github.com/go-gitea/gitea/pull/25401/commits/872b9f9fbe8ac917f53942f43a8e794a56a0ed83.

silverwind commented:
Ready again. I've moved the function and tests to separate files and restored all previous tests.

The problem with the `style="color:rgb(255,255,255)"` was that a `ansi_up` instance is stateful so when one test opened and not closed the color tag, it remembered that for future invocations. I removed this issue now by creating a fresh instance on every function call.

The seemingly odd `AnsiUp.default || AnsiUp` seems necessary so that the module correctly imports in both vitest and the browser, which I think is an issue related to their legacy AMD wrapper that the module features.
